src/Cryptol/Eval/Reference.lhs

```diff
@@ -128,150 +161,85 @@ terms by providing an evaluator to an appropriate `Value` type.
+>   | VRecord [(Ident, E Value)] -- ^ @ { .. } @ records
+>   | VFun (E Value -> E Value)  -- ^ functions
+>   | VPoly (TValue -> E Value)  -- ^ polymorphic values (kind *)
+>   | VNumPoly (Nat' -> E Value) -- ^ polymorphic values (kind #)
 
 Operations on Values
 --------------------
 
 > -- | Destructor for @VBit@.
-> fromVBit :: Value -> Either EvalError Bool
+> fromVBit :: Value -> Bool
 > fromVBit (VBit b) = b
 > fromVBit _        = evalPanic "fromVBit" ["Expected a bit"]
 >
 > -- | Destructor for @VInteger@.
-> fromVInteger :: Value -> Either EvalError Integer
+> fromVInteger :: Value -> Integer
 > fromVInteger (VInteger i) = i
 > fromVInteger _            = evalPanic "fromVInteger" ["Expected an integer"]
 >
 > -- | Destructor for @VRational@.
-> fromVRational :: Value -> Either EvalError Rational
+> fromVRational :: Value -> Rational
 > fromVRational (VRational i) = i
 > fromVRational _             = evalPanic "fromVRational" ["Expected a rational"]
 >
-> fromVFloat :: Value -> Either EvalError BigFloat
-> fromVFloat = fmap bfValue . fromVFloat'
+> fromVFloat :: Value -> BigFloat
+> fromVFloat = bfValue . fromVFloat'
 
-> fromVFloat' :: Value -> Either EvalError BF
+> fromVFloat' :: Value -> BF
 > fromVFloat' v =
 >   case v of
 >     VFloat f -> f
 >     _ -> evalPanic "fromVFloat" [ "Expected a floating point value." ]
 
-
-
-
 >
 > -- | Destructor for @VList@.
-> fromVList :: Value -> [Value]
+> fromVList :: Value -> [E Value]
 > fromVList (VList _ vs) = vs
 > fromVList _            = evalPanic "fromVList" ["Expected a list"]
 >
 > -- | Destructor for @VTuple@.
-> fromVTuple :: Value -> [Value]
+> fromVTuple :: Value -> [E Value]
 > fromVTuple (VTuple vs) = vs
 > fromVTuple _           = evalPanic "fromVTuple" ["Expected a tuple"]
 >
 > -- | Destructor for @VRecord@.
-> fromVRecord :: Value -> [(Ident, Value)]
+> fromVRecord :: Value -> [(Ident, E Value)]
 > fromVRecord (VRecord fs) = fs
 > fromVRecord _            = evalPanic "fromVRecord" ["Expected a record"]
 >
 > -- | Destructor for @VFun@.
-> fromVFun :: Value -> (Value -> Value)
+> fromVFun :: Value -> (E Value -> E Value)
 > fromVFun (VFun f) = f
 > fromVFun _        = evalPanic "fromVFun" ["Expected a function"]
 >
-> -- | Destructor for @VPoly@.
-> fromVPoly :: Value -> (TValue -> Value)
-> fromVPoly (VPoly f) = f
-> fromVPoly _         = evalPanic "fromVPoly" ["Expected a polymorphic value"]
->
-> -- | Destructor for @VNumPoly@.
-> fromVNumPoly :: Value -> (Nat' -> Value)
-> fromVNumPoly (VNumPoly f) = f
-> fromVNumPoly _            = evalPanic "fromVNumPoly" ["Expected a polymorphic value"]
->
+
+ -- | Destructor for @VPoly@.
+ fromVPoly :: Value -> (TValue -> E Value)
+ fromVPoly (VPoly f) = f
+ fromVPoly _         = evalPanic "fromVPoly" ["Expected a polymorphic value"]
+
+ -- | Destructor for @VNumPoly@.
+ fromVNumPoly :: Value -> (Nat' -> E Value)
+ fromVNumPoly (VNumPoly f) = f
+ fromVNumPoly _            = evalPanic "fromVNumPoly" ["Expected a polymorphic value"]
+
```

Comment:
If we don't need these functions anymore, we should just remove them completely; these are formatted rather badly in the generated `Semantics.pdf`.

Reply:
Agreed.  Seems like they aren't necessary.